docs/dev/testing.rst

```diff
@@ -0,0 +1,306 @@
+is configured with the ``tox.ini`` file in the main repository directory. There
+are several different "test environments" defined, to test different aspects
+of the software, or to perform other actions like building the documentation.
+To see a list of the available environments with short descriptions in the PUDL
+repository run:
+
+.. code-block:: console
+
+    $ tox -av
+
+    default environments:
+    ci               -> Run all continuous integration (CI) checks & generate test coverage.
+
+    additional environments:
+    flake8           -> Run the full suite of flake8 linters on the PUDL codebase.
+    pre_commit       -> Run git pre-commit hooks not covered by the other linters.
+    bandit           -> Check the PUDL codebase for common insecure code patterns.
+    linters          -> Run the pre-commit, flake8 and bandit linters.
+    doc8             -> Check the documentation input files for syntactical correctness.
+    docs             -> Remove old docs output and rebuild HTML from scratch with Sphinx
+    unit             -> Run all the software unit tests.
+    ferc1_solo       -> Test whether FERC 1 can be loaded into the PUDL database alone.
+    integration      -> Run all software integration tests and process a full year of data.
+    validate         -> Run all data validation tests. This requires a complete PUDL DB.
+    ferc1_schema     -> Verify FERC Form 1 DB schema are compatible for all years.
+    full_integration -> Run ETL and integration tests for all years and data sources.
+    full             -> Run all CI checks, but for all years of data.
+    build            -> Prepare Python source and binary packages for release.
+    testrelease      -> Do a dry run of Python package release using the PyPI test server.
+    release          -> Release the PUDL package to the production PyPI server.
+
+Continuous Integration Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Our default tox test environment is ``ci`` -- which includes all of the tests
+that will be run in continuous integration using a `GitHub Action
+<https://github.com/features/actions>`__. You should run these tests before
+pushing code to the repository or making a pull request. Because it's the
+default test environment, it will be run if you call Tox without any
+arguments:
+
+.. code-block:: console
+
+    $ tox
+
+This is equivalent to:
+
+.. code-block:: console
+
+    $ tox -e ci
+
+If the dependencies have been changed, or you recently ran the tests while on
+another branch of the repository with other dependencies, you should tell Tox
+to rebuild the software environment it operates in with the ``-r`` (aka
+``--recreate``) flag:
+
+.. code-block:: console
+
+    $ tox -re ci
+
+In addition to running the ``unit`` and ``integration`` tests, the CI test
+environment lints the code and documentation input files, and uses Sphinx
+build the documentation. It also generates a test coverage report. Running
```

Comment:
I think it should be `and uses Sphinx **to** build the documentation`

Reply:
Fixed.